src/client/dfuse/il/int_posix.c

```diff
@@ -377,12 +398,55 @@ ioil_fini(void)
 			ioil_shrink_pool(pool);
 	}
 
-	if (ioil_iog.iog_daos_init)
+	if (ioil_iog.iog_daos_init) {
+		int i;
+
+		/** destroy EQs created by threads */
+		for (i = 0; i < ioil_iog.iog_eq_count; i++)
+			daos_eq_destroy(ioil_iog.iog_eqs[i], 0);
+		/** destroy main thread eq */
+		if (daos_handle_is_valid(ioil_iog.iog_main_eqh))
+			daos_eq_destroy(ioil_iog.iog_main_eqh, 0);
 		daos_fini();
+	}
 	ioil_iog.iog_daos_init = false;
 	daos_debug_fini();
 }
 
+int
+ioil_get_eqh(daos_handle_t *eqh)
+{
+	int rc;
+
+	if (daos_handle_is_valid(ioil_eqh)) {
+		*eqh = ioil_eqh;
+		return 0;
+	}
+
+	/** No EQ support requested */
+	if (ioil_iog.iog_eq_count_max == 0)
+		return -1;
+
+	rc = pthread_mutex_lock(&ioil_iog.iog_lock);
+	/** create a new EQ if the EQ pool is not full; otherwise round robin EQ use from pool */
+	if (ioil_iog.iog_eq_count >= ioil_iog.iog_eq_count_max) {
+		ioil_eqh = ioil_iog.iog_eqs[ioil_iog.iog_eq_idx ++];
```

Comment:
Shall we move Line 433 "ioil_eqh = ioil_iog.iog_eqs[ioil_iog.iog_eq_idx ++];" after line 435?

Reply:
can you explain why that is needed?

Reply:
Line 433 uses the value "ioil_iog.iog_eq_idx" before the initialization in line 435. The code should be fine since static variables "static struct ioil_global ioil_iog" are set zero anyway. If we rely on the default zero value of ioil_iog.iog_eq_idx, there is no need of line 435 then.

Reply:
right, that is my assumption. static global variables are already initialized to 0. So this is not needed IMO

Reply:
I see. You can remove line 435 either in this PR or in future PR.

Reply:
Yes. You are right.